### PR TITLE
feat: add typescript support using Bun runtime

### PR DESCRIPTION
**Pull request by ryan-gang**

### CODE REVIEW

**Comment by ryan-gang** [comment]
Implemented the starter using `Deno.serve` initially. (https://deno.land/api@v1.42.4?s=Deno.serve)
But it kept on failing our anti-cheat. 
Turns out the server automatically adds the `date` header, which we check in the anti-cheat stage.

Wanted to document it for later.

**Review by rohitpaulk — CHANGES_REQUESTED** [review]
Added notes in the Git PR, dockerfile shouldn't be using app code

File: compiled_starters/typescript/app/main.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import * as net from 'net';
+
+// You can use print statements as follows for debugging, they'll be visible when running tests.
```
Comment (rohitpaulk) [inline compiled_starters/typescript/app/main.ts:3]: Let's move this to just above the "// uncomment this" block (it's more likely to get noticed that way + the diff looks cleaner)

File: course-definition.yml [hunk 2]
```diff
@@ -76,6 +76,8 @@ languages:
     release_status: "beta"
   - slug: "ruby"
     release_status: "beta"
+  - slug: "typescript"
+    release_status: "beta"
```
Comment (rohitpaulk) [inline course-definition.yml:80]: Would recommend doing `alpha` first and trying this out btw - at least for the first challenge that we deploy that uses Bun. After that it's okay to do beta directly if you're confident

File: solutions/typescript/01-connect-to-port/code/codecrafters.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Set this to true if you want debug logs.
+#
+# These can be VERY verbose, so we suggest turning them off
+# unless you really need them.
+debug: false
+
+# Use this to change the TypeScript version used to run your code
+# on Codecrafters.
+#
+# Available versions: deno-1.42
```
Comment (rohitpaulk) [inline solutions/typescript/01-connect-to-port/code/codecrafters.yml:10]:
```suggestion
# Available versions: bun-1.1
```

**Pull request merged**